src/Administration/Resources/app/administration/src/module/sw-users-permissions/page/sw-sso-users-permission-user-detail/index.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * @internal
+ * @sw-package framework
+ */
+import template from './sw-sso-users-permission-user-detail.html.twig';
+import './sw-sso-users-permissions-user-detail.scss';
+
+const { Mixin } = Shopware;
+const { Criteria } = Shopware.Data;
+
+const MODE = {
+    VIEW: 'view',
+    EDIT: 'edit',
+    CREATE: 'create',
+};
+
+// eslint-disable-next-line sw-deprecation-rules/private-feature-declarations
+export default {
+    template,
+
+    inject: [
+        'acl',
+        'repositoryFactory',
+        'integrationService',
+        'userService',
+    ],
+
+    mixins: [
+        Mixin.getByName('notification'),
+        Mixin.getByName('salutation'),
+    ],
+
+    shortcuts: {
+        'SYSTEMKEY+S': 'onSave',
+        ESCAPE: 'onCancel',
+    },
+
+    created() {
+        this.userId = this.$route.params.id;
+        this.loadRepositories();
```

Comment:
please use promise.all (make methods async), so we avoid race conditons and this way you can also get rid of the complex loading state logic and just use a boolean
```suggestion
await Promise.all([
        this.loadUser(),
        this.loadCurrentUser(),
        this.loadLanguages(),
    ]);
```

Reply:
Good suggestion. I discovered other code smells in this component as well, which led me to refactor a lot of it and simplify it in the process 😅